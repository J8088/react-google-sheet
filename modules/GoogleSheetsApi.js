import React from 'react'
import { GoogleApi } from '@lourd/react-google-api'

const GoogleSheetsApi = ({
  scopes = ['https://www.googleapis.com/auth/spreadsheets.readonly'],
  ...props
}) => (
  <GoogleApi
    scopes={scopes}
    discoveryDocs={['https://sheets.googleapis.com/$discovery/rest?version=v4']}
    {...props}
  />
)

export default GoogleSheetsApi
