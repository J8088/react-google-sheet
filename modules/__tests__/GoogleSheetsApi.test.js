import React from 'react'
import { mount } from 'enzyme'

import GoogleSheetsApi from '../GoogleSheetsApi'

it('renders the GoogleApi with the sheets discovery doc provided and a default scope and passes all props through', () => {
  const result = mount(
    <GoogleSheetsApi clientId="foo" apiKey="bar" cool="ðŸ¤ ">
      {null}
    </GoogleSheetsApi>,
  )
  expect(result).toMatchSnapshot('default case')
})

it('accepts an array of scopes', () => {
  const result = mount(
    <GoogleSheetsApi clientId="foo" apiKey="bar" scopes={['baz', 'yup']}>
      {null}
    </GoogleSheetsApi>,
  )
  expect(result).toMatchSnapshot('custom scopes')
})
