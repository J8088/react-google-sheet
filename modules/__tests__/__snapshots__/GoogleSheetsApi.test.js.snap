// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts an array of scopes: custom scopes 1`] = `
<GoogleSheetsApi
  apiKey="bar"
  clientId="foo"
  scopes={
    Array [
      "baz",
      "yup",
    ]
  }
>
  <GoogleApi
    apiKey="bar"
    clientId="foo"
    discoveryDocs={
      Array [
        "https://sheets.googleapis.com/$discovery/rest?version=v4",
      ]
    }
    scopes={
      Array [
        "baz",
        "yup",
      ]
    }
  >
    <GoogleApiProvider
      value={
        Object {
          "authorize": [Function],
          "client": null,
          "error": null,
          "loading": true,
          "signedIn": false,
          "signout": [Function],
        }
      }
    />
  </GoogleApi>
</GoogleSheetsApi>
`;

exports[`renders the GoogleApi with the sheets discovery doc provided and a default scope and passes all props through: default case 1`] = `
<GoogleSheetsApi
  apiKey="bar"
  clientId="foo"
  cool="ðŸ¤ "
>
  <GoogleApi
    apiKey="bar"
    clientId="foo"
    cool="ðŸ¤ "
    discoveryDocs={
      Array [
        "https://sheets.googleapis.com/$discovery/rest?version=v4",
      ]
    }
    scopes={
      Array [
        "https://www.googleapis.com/auth/spreadsheets.readonly",
      ]
    }
  >
    <GoogleApiProvider
      value={
        Object {
          "authorize": [Function],
          "client": null,
          "error": null,
          "loading": true,
          "signedIn": false,
          "signout": [Function],
        }
      }
    />
  </GoogleApi>
</GoogleSheetsApi>
`;
